app/build.gradle

```diff
@@ -68,4 +71,29 @@ dependencies {
     androidTestImplementation 'androidx.compose.ui:ui-test-junit4'
     debugImplementation 'androidx.compose.ui:ui-tooling'
     debugImplementation 'androidx.compose.ui:ui-test-manifest'
+    implementation 'androidx.work:work-runtime:2.9.0'
+
+    implementation "androidx.navigation:navigation-common-ktx:2.7.7"
+    implementation "androidx.lifecycle:lifecycle-runtime-compose:2.7.0"
+
+    //navigation dependencies
+    def nav_version = "2.7.7"
+    def hilt_version = "2.48"
+    // Kotlin
+    implementation "androidx.navigation:navigation-fragment-ktx:$nav_version"
+    implementation "androidx.navigation:navigation-ui-ktx:$nav_version"
+
+    // Jetpack Compose Integration
+    implementation "androidx.navigation:navigation-compose:$nav_version"
+
+    //hilt dependencies
+    implementation 'androidx.hilt:hilt-work:1.1.0'
+    implementation "com.google.dagger:hilt-android:$hilt_version"
+    kapt "com.google.dagger:hilt-compiler:$hilt_version"
```

Comment:
I recommend to use KSP (Kotlin Symbol Processing), which is a faster alternative to kapt. Both Dagger and Room support that.

I explain it here: https://www.youtube.com/watch?v=WVhgUGyUnhQ